### PR TITLE
NOTIF-496 Return 400 when validating invalid b/a/et triplet

### PR DESCRIPTION
**Pull request by gwenneg**
This is part 2.

See also [part 1](https://github.com/RedHatInsights/notifications-backend/pull/1519).

### CODE REVIEW

File: src/main/java/com/redhat/cloud/notifications/GwResource.java [hunk 1]
```diff
@@ -73,6 +74,13 @@ public GwResource(MeterRegistry registry) {
     public Response forward(@NotNull @Valid RestAction ra) {
         receivedActions.increment();
 
+        try {
+            restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
+        } catch (BadRequestException e) {
+            // The following line is required to forward the HTTP 400 error message.
+            return Response.status(BAD_REQUEST).entity(e.getMessage()).build();
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/GwResource.java:81]: Isn't this automatically forwarded by quarkus if not catched? (the `BadRequest`)

Comment (gwenneg) [reply src/main/java/com/redhat/cloud/notifications/GwResource.java:81]: The status is forwarded but the message is not, so this is necessary. We have the same thing in notifications.

File: src/main/java/com/redhat/cloud/notifications/RestValidationClient.java [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+@ApplicationScoped // This scope is required to mock the client during tests.
 public interface RestValidationClient {
 
     @GET
     @Path("/baet")
     @Retry(maxRetries = 5)
-    @Produces(MediaType.APPLICATION_JSON)
-    Response isBundleApplicationEventTypeTripleValid(@QueryParam("bundle") String bundle, @QueryParam("application") String application, @QueryParam("eventType") String eventType);
+    @Produces(TEXT_PLAIN)
+    Response validate(@QueryParam("bundle") String bundle, @QueryParam("application") String application, @QueryParam("eventType") String eventType);
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/RestValidationClient.java:24]: I think that in the future we should always use a JSON response.

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**